akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -665,7 +674,23 @@ object Flow {
    * '''Cancels when''' downstream cancels
    */
   def lazyFutureFlow[I, O, M](create: () => Future[Flow[I, O, M]]): Flow[I, O, Future[M]] =
-    Flow.fromGraph(new LazyFlow(_ => create()))
+    /*Flow.fromGraph(new LazyFlow(_ => create()))*/ Flow[I]
+      .flatMapPrefixMat(1) {
+        case Seq(a) =>
+          implicit val ec = akka.dispatch.ExecutionContexts.parasitic
+          val f: Flow[I, O, Future[M]] =
+            futureFlow(create().map(Flow[I].prepend(Source.single(a)).viaMat(_)(Keep.right)))
+          f
+        case Seq() =>
+          val f: Flow[I, O, Future[M]] = Flow[I]
+            .asInstanceOf[Flow[I, O, NotUsed]]
```

Comment:
Why is that needed?

Reply:
could also be done by:
`Flow[I].map[O](_ => ???)`
which one do you prefer?

Reply:
> Why is that needed?

`O` can't be inferred otherwise.

Reply:
Yes, agreed.